### ISSUE TITLE
a very confusing paragraph in the beginner's guide

### ISSUE THREAD

**Issue by nyck33** [Unity-Technologies/ml-agents#2064]
> What we learned
> From setting up the training scene, we learned that parallelizing training instead of just repeating the same situation several times makes the training more solid, as the agents have more data to work with and learn from. To do this, it is useful to set-up a few Heuristic agents to perform simple actions that help the agents learn, without leading them into making wrong assumptions. For instance, we set-up a simple Heuristic script to make our agents attack randomly, to provide our training agents with an example of what happens when they get receive damage.

1.  How did you parallelize training?  Please be specific and provide references to code snippets by line if possible.

2.  What do you mean by, "set up a few heuristic agents"?  Did you modify the reward function within an RL algorithm to get the random attacking behavior?  Please explain.  

3.  "...into making wrong assumptions"  Please provide some examples of wrong assumptions in this context.

4.  "...we set-up a simple Heuristic script to make our agents attack randomly, to provide our training agents with an example of what happens when they get receive damage."  You seem to be talking about two agents, one attacking and one getting attacked.  Can you clarify which one the adversary is and which is the agent we are training?  Or are they both RL agents being trained to do different things against each other?

**Comment by vincentpierre** [comment]
Hi, 
Within the same scene, there are multiple training agents that are collecting data and learning. There are also opponents to the training agents that use a heuristic in order to challenge the training agent.
1. We call this parallel training because there are multiple agents training at the same time. This is the same as in Balance Ball : there are 12 agents learning at once. 
2. The heuristic agents are not training, they are only providing an opponent to the agents that are training. They use a heuristic brain that is user implemented (in this case they act randomly).
3. If the learning agent was facing a heuristic coded by a developer, it could maybe exploit some of the defects in the heuristic. For instance, if the developer said that the agent must attack only every 3 seconds, then the agent would learn to disengage every 3 seconds. This behavior would not work against another heuristic or a human player. Using a random heuristic prevents that. There are probably better ways to approach this problem like having an opponent that is a past version of the learning agent (self-play).
4. The heuristic agent is fighting the learning agent to train it, this is a 1v1 game I think.

**Comment by nyck33** [comment]
Noted with thanks.

In the picture below, in Room 4, you would have the agent learning against a human?  

![image](https://user-images.githubusercontent.com/25189545/58142561-ca60c900-7c71-11e9-8b17-9215c2849840.png)

**Comment by vincentpierre** [comment]
I think that last room is for testing, if you trained against a human, you will not be able to speed up the game.

**Comment by roboserg** [comment]
@vincentpierre  Since this issue concerns parallelization I wanted to what is the best way to set the number of training areas within one scene AND balance it with several concurrent Unity instances? 

For example, I have an i7 with 4 cores and 4 HT, so 8 virtual cores.

If I set up 16 training areas within one unity instance my overall CPU load is 80%-85%. Should I still do a second unity instance with an extra 16 areas? I don't quite understand how to make the best use of the multiple areas within one scene vs multiple unity instances. 

Do I set up so many scenes that my CPU load is 50% and then do a second instance so that the total CPU load is 100%? Or do I just do as many training areas as possible in one scene to reach 100%? In you blog post you were specifically saying multiple instances make training faster on "home" PC's.
